provider/devices/tizen.go

```diff
@@ -0,0 +1,353 @@
+		return
+	}
+
+	device.OS = "tizen"
+
+	go startAppium(device, &wg)
+	go checkAppiumUp(device)
+
+	select {
+	case <-device.AppiumReadyChan:
+		logger.ProviderLogger.LogInfo("tizen_device_setup", fmt.Sprintf("Successfully started Appium for device `%v` on port %v", device.UDID, device.AppiumPort))
+		break
+	case <-time.After(30 * time.Second):
+		logger.ProviderLogger.LogError("tizen_device_setup", fmt.Sprintf("Did not successfully start Appium for device `%v` in 60 seconds", device.UDID))
+		ResetLocalDevice(device, "Appium did not start within the expected time.")
+		return
+	}
+
+	device.ProviderState = "live"
+	wg.Wait()
+}
+
+func getTizenTVHost(tvID string) (string, error) {
+	// Check if the hostWithPort is in the format HOST_IP:PORT
+	if matched, _ := regexp.MatchString(`^([0-9]{1,3}\.){3}[0-9]{1,3}:\d+$`, tvID); matched {
+		host := strings.Split(tvID, ":")[0]
+		return host, nil
+	} else {
+		return "", fmt.Errorf("invalid format for host: %s", tvID)
+	}
+}
+
+func getTizenTVInfo(device *models.Device) error {
+	tvHost, err := getTizenTVHost(device.UDID)
+	if err != nil {
+		return fmt.Errorf("failed to get TV host - %s", err)
+	}
+
+	url := fmt.Sprintf("http://%s:8001/api/v2/", tvHost)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return fmt.Errorf("failed to get TV info - %s", err)
+	}
+	defer resp.Body.Close()
+
+	var tvInfo TizenTVInfo
+	if err := json.NewDecoder(resp.Body).Decode(&tvInfo); err != nil {
+		return fmt.Errorf("failed to decode TV info - %s", err)
+	}
+
+	// Atualizar informações do device
+	device.Name = tvInfo.Device.Name
+	device.HardwareModel = tvInfo.Device.ModelName
+	device.OSVersion = tvInfo.Version
+	device.IPAddress = tvInfo.Device.IP
+	device.DeviceAddress = device.UDID
+
+	// Extrair dimensões da resolução
+	if tvInfo.Device.Resolution != "" {
+		dimensions := strings.Split(tvInfo.Device.Resolution, "x")
+		if len(dimensions) == 2 {
+			device.ScreenWidth = dimensions[0]
+			device.ScreenHeight = dimensions[1]
+		}
+	}
+
+	return nil
+}
+
+// connectTizenDevice establishes a connection to a Tizen device using sdb connect
+func connectTizenDevice(deviceIP string) error {
+	logger.ProviderLogger.LogInfo("tizen_connection", fmt.Sprintf("Attempting to connect to Tizen device at %s", deviceIP))
+
+	cmd := exec.Command("sdb", "connect", deviceIP)
+	output, err := cmd.CombinedOutput()
+
+	if err != nil {
+		logger.ProviderLogger.LogError("tizen_connection", fmt.Sprintf("Failed to connect to Tizen device %s - %s. Output: %s", deviceIP, err, string(output)))
+		return fmt.Errorf("failed to connect to Tizen device %s: %s", deviceIP, err)
+	}
+
+	logger.ProviderLogger.LogInfo("tizen_connection", fmt.Sprintf("Successfully connected to Tizen device %s. Output: %s", deviceIP, string(output)))
+	return nil
+}
+
+// isTizenDeviceConnected checks if a Tizen device is currently connected using sdb devices
+func isTizenDeviceConnected(deviceIP string) bool {
```

Comment:
Isn't this quite similar to `getConnectedDevicesTizen()`?

Reply:
I refactored isTizenDeviceConnected() to use getConnectedDevicesTizen(). Done in [commit](https://github.com/shamanec/GADS/pull/208/commits/9da3af4c0ebe71de3b3f2268cf3e7102cf0846c3)